File: sdk/dotnet-sdk/src/VeritySDK/Utils/DbcUtil.cs
using System;

namespace VeritySDK.Utils
{
    /// <summary>
    /// Utilities for design by contract. Allows for common pre-condition and post-condition checking.
    /// </summary>
    public class DbcUtil
    {
        /// <summary>
        /// Checks and throws an ArgumentException if the given object is null
        /// </summary>
        /// <param name="arg">any object that can be null</param>
        public static void requireNotNull(object arg)
        {
            requireNotNull(arg, "ARG");
        }

        /// <summary>
        /// Checks and throws an ArgumentException if the given object is null
        /// </summary>
        /// <param name="arg">any object that can be null</param>
        /// <param name="argName">name of the argument being checked, used in exception message if null</param>
        public static void requireNotNull(object arg, string argName)
        {
            require(arg != null, "required that " + argName + " must NOT be null");
        }

        /// <summary>
        /// Checks that given boolean is true, throws an ArgumentException if false
        /// </summary>
        /// <param name="requirement">requirement testable requirement</param>
        public static void require(bool requirement)
        {
            require(requirement, "");
        }

        /// <summary>
        /// Checks that given boolean is true, throws an ArgumentException if false
        /// </summary>
        /// <param name="requirement">requirement testable requirement</param>
        /// <param name="msg">string that is used to build exception message</param>
        public static void require(bool requirement, string msg)
        {
            if (!requirement)
            {
                throw new ArgumentException("requirement failed: " + msg);
            }
        }

        /// <summary>
        /// Checks and throws an ArgumentException if the given object is null
        /// </summary>
        /// <param name="arg">string argument which needs to be checked</param>
        /// <param name="argName">name of the argument being checked, used in exception message</param>
        public static void requireStringNotNullOrEmpty(string arg, string argName)
        {
            require(!String.IsNullOrEmpty(arg), "required that " + argName + " must NOT be null or empty string");
        }

        /// <summary>
        /// Checks and throws an ArgumentException if the given array is null or contains null element
        /// </summary>
        /// <param name="array">array which needs to be checked</param>
        /// <param name="argName">name of the argument being checked, used in exception message</param>
        public static void requireArrayNotContainNull(object[] array, string argName)
        {
            requireNotNull(array, argName);
            foreach (object x in array)
                require(x != null, "required that elements of " + argName + " must NOT be null");
        }
    }
}

File: sdk/nodejs-sdk/src/utils/DbcUtil.js
'use strict'

/**
 * Utilities for design by contract. Allows for common pre-condition and post-condition checking.
 */
class DbcUtil {
  /**
   * Checks that given requirement is true, throws an Error if false
   * @param {Boolean} requirement - testable requirement
   * @param {String} msg - string that is used to build exception message
   */
  static require (requirement, msg) {
    if (!requirement) {
      throw new Error('requirement failed: ' + msg)
    }
  }

  /**
   * Checks and throws an Error if the given object is null
   * @param {Object} arg - any object that can be null
   * @param {String} argName - name of the argument being checked, used in exception message if null
   */
  static requireNotNull (arg, argName) {
    this.require(arg != null, 'required that ' + argName + ' must NOT be null')
  }

  /**
   * Checks and throws an Error if the given string is null or empty
   * @param {String} arg - String argument which needs to be checked
   * @param {String} argName - name of the argument being checked, used in exception message if null or empty
   */
  static requireStringNotNullOrEmpty (arg, argName) {
    this.require(arg != null && arg.trim(), 'required that ' + argName + ' must NOT be null or empty')
  }

  /**
   * Checks and throws an Error if the given array is null or contains null element
   * @param {Array} array - Array argument which needs to be checked
   * @param {String} argName - name of the argument being checked, used in exception message
   */
  static requireArrayNotContainNull (array, argName) {
    this.requireNotNull(array, argName)
    array.forEach(function (x) {
      DbcUtil.require(x != null, 'required that elements of ' + argName + ' must NOT be null')
    })
  }
}

module.exports = DbcUtil
